manticore/core/smtlib/solver.py

```diff
@@ -352,34 +314,39 @@ def _is_sat(self) -> bool:
 
         :return: whether current state is satisfiable or not.
         """
-        logger.debug("Solver.check() ")
         start = time.time()
-        self._send("(check-sat)")
-        status = self._recv()
+        self._smtlib.send("(check-sat)")
+        status = self._smtlib.recv()
         logger.debug("Check took %s seconds (%s)", time.time() - start, status)
         if status not in ("sat", "unsat", "unknown"):
             raise SolverError(status)
         if consts.defaultunsat:
             if status == "unknown":
                 logger.info("Found an unknown core, probably a solver timeout")
                 status = "unsat"
-
         if status == "unknown":
             raise SolverUnknown(status)
-
-        is_sat = status == "sat"
-        if not is_sat:
-            check_memory_usage()
-            check_disk_usage()
-        return is_sat
+        return status == "sat"
 
     def _assert(self, expression: Bool):
         """Auxiliary method to send an assert"""
-        assert isinstance(expression, Bool)
         smtlib = translate_to_smtlib(expression)
-        self._send("(assert %s)" % smtlib)
-
-    def _getvalue(self, expression):
+        self._smtlib.send(f"(assert {smtlib})")
+
+    # Union[Variable, int, bool, bytes]
+    def __getvalue_bv(self, expression_str):
```

Comment:
Should that type hint be moved into the function argument?
```suggestion
    def __getvalue_bv(self, expression_str: Union[Variable, int, bool, bytes]) -> int:
```